### ISSUE TITLE
question - Mapie for zero inflated regressors

### ISSUE THREAD

**Issue by SSMK-wq** [scikit-learn-contrib/MAPIE#240]
I am currently working on regression problem which mostly has lot of zeros (for 80% of the data) and 20% of non-zeros. So, we are using the `scikit-lego` package which has `Zero Inflated Regressor` as shown [https://scikit-lego.netlify.app/meta.html#Zero-Inflated-Regressor](here)

My code looks like below

```
clf = GridSearchCV(
    estimator=ExtraTreesClassifier(),
    param_grid={
        'n_estimators': [100,200,300,400,500],
        'bootstrap':[True, False],
        'max_features': ['sqrt','log2',None],
        'max_depth':[2,4,6,8,None],
        'class_weight':['balanced']
    },
    scoring='roc_auc',
)

reg = GridSearchCV(
    estimator=RandomForestRegressor(),
    param_grid={
        'n_estimators': [100,200,300,400,500],
        'bootstrap':[True, False],
        'max_features': ['sqrt','log2',None],
        'max_depth':[2,4,6,8,None],
    },
    scoring = 'neg_mean_squared_error',
)
       
zir = ZeroInflatedRegressor(
    classifier=clf,
    regressor=reg,
)

estimator = zir.fit(X_train_poly,y_train)
mapie = MapieRegressor(estimator=estimator, cv="prefit").fit(X_cal_poly, y_cal)
y_test_interval_pred_mapie = mapie.predict(X_test_poly, alpha=.10)[1].reshape(-1,2)
```
But I see around around 40% of my observations fall outside the `low and high` limits.

Additionally, I also see that the intervals are very wide. Is there anyway to make intervals narrower and improve the accuracy of placing the actual value within a narrow interval?

**Expected behavior**
Actual value should be within a narrow interval and low miscoverage

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser chrome 
 - MAPIE Version - 0.5.0

**Additional context**
How can we improve the accuracy of Mapie Regressor? Any best practice to improve the interval width and reduce the miscoverage rate?

While this is not a bug, I could not select any other option to pose question. So, chose this option (others were feature request).

**Comment by LacombeLouis** [comment]
Hey @SSMK-wq, thank you for your question. Conformal predictions rely on the assumption of exchangeability of the dataset (so a drift of your in your train/calibration/test set could lead to under-coverage). If this does not hold, then it could explain why you would have coverage lower than your wanted target coverage. 

Additionally, feel free to try the different conformal prediction methods as they might have better coverages or interval widths, you can see a summary of them in the [documentation](https://mapie.readthedocs.io/en/latest/theoretical_description_regression.html) and that you can implement using the `method` and `cv` attributes.

- "naive": (method="naive"),
- "jackknife": (method="base", cv=-1),
- "jackknife_plus": (method="plus", cv=-1),
- "jackknife_minmax": (method="minmax", cv=-1),
- "cv": (method="base", cv=10),
- "cv_plus": (method="plus", cv=10),
- "cv_minmax": (method="minmax", cv=10),
- "jackknife_plus_ab": (method="plus", cv=Subsample(n_resamplings=50)),
- "jackknife_minmax_ab": (method="minmax", cv=Subsample(n_resamplings=50))

I hope this helps!
